website/js/d3Lollipop.js

```diff
@@ -78,21 +111,30 @@ var D3Lollipop = React.createClass({
         var newObj = {category: oldObj.Pathogenicity_expert, coord: chrCoordinate, value: 1, oldData: obj};
         return newObj;
     },
-
-     componentDidMount: function() {
-        var {data, brcakey, onRowClick, ...opts} = this.props;
-        var subSetData = data.map(this.filterAttributes);
+    componentDidMount: function() {
+        this.startSpinner();
+        var {dataIsEmpty, data, brcakey, onRowClick, ...opts} = this.props;
+        // Redraw the chart when brcakey has changed.
         var d3svgBrcaRef = React.findDOMNode(this.refs.d3svgBrca);
+        var subSetData = data.map(this.filterAttributes);
         var domainBRCA = JSON.parse(brca12JSON[brcakey].brcaDomainFile);
-        this.cleanupBRCA = d3Lollipop.drawStuffWithD3(d3svgBrcaRef, subSetData, domainBRCA, brcakey, onRowClick);
+        if (data.length !== 0 || dataIsEmpty) {
```

Comment:
this should be cleaned up or there should at least be a comment here to explain what's going on -- it's particularly confusing because it seems like  `if (data.length !== 0 or dataIsEmpty)` would always be true since they're opposites of each other. 

what it's really doing is stopping the spinner as soon as the data is finished loading, and it's only drawing the lollipop chart if the data is not empty.

Reply:
I've tried the following which I believe is logically equivalent, it behaves the same way. I'm still confused how data.length !== 0 and dataIsEmpty are NOT mutually exclusive (aka why are they not logically opposite)?
`if (data.length !== 0) {`
`           this.cleanupBRCA = d3Lollipop.drawStuffWithD3(d3svgBrcaRef, subSetData, domainBRCA, brcakey, onRowClick);`
`            // Prevent the spinner from showing up when changing the gene tab`
`            this.stopSpinner();`
`      } else if (dataIsEmpty) {`
`            this.stopSpinner();`
 `};`

Reply:
yea it's extremely confusing...

`data.length !== 0` is there to tell us when the data is done loading. the problem is if the data finishes loading but it's length is actually 0, in which case data.length !== 0 is still false. In the case that data is loaded and it's length is 0, dataIsEmpty is set to True (it defaults to false), so we still make a call to this.stopSpinner, but don't load the chart.

Make sense?